### PR TITLE
Fix credentials throwing error on `kedro catalog resolve`

### PR DESCRIPTION
**Pull request by merelcht**
## Description
Feed credentials to catalog in `kedro catalog resolve` command.

## Development notes
This bug was reported by a user on Slack. In short the `kedro catalog resolve` command uses 
```python
DataCatalog.from_config(catalog=catalog_config)
```
But this is missing the `credentials` argument. If a user has `credentials` in a dataset that also uses dataset factories an error is thrown:

```
KeyError: "Unable to find credentials 'sql_tb_credentials': check your data catalog and credentials configuration. See https://kedro.readthedocs.io/en/stable/kedro.io.DataCatalog.html for an example."
```

## Developer Certificate of Origin
We need all contributions to comply with the [Developer Certificate of Origin (DCO)](https://developercertificate.org/). All commits must be signed off by including a `Signed-off-by` line in the commit message. [See our wiki for guidance](https://github.com/kedro-org/kedro/wiki/Guidelines-for-contributing-developers/).

If your PR is blocked due to unsigned commits, then you must follow the instructions under "Rebase the branch" on the GitHub Checks page for your PR. This will retroactively add the sign-off to all unsigned commits and allow the DCO check to pass.

## Checklist

- [ ] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [x] Signed off each commit with a [Developer Certificate of Origin (DCO)](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/managing-repository-settings/managing-the-commit-signoff-policy-for-your-repository)
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [x] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [x] Added tests to cover my changes
- [ ] Checked if this change will affect Kedro-Viz, and if so, communicated that with the Viz team

### CODE REVIEW

File: kedro/framework/cli/catalog.py [hunk 1]
```diff
@@ -233,7 +233,10 @@ def resolve_patterns(metadata: ProjectMetadata, env: str) -> None:
     context = session.load_context()
 
     catalog_config = context.config_loader["catalog"]
-    data_catalog = DataCatalog.from_config(catalog_config)
+    credentials_config = context.config_loader.get("credentials", None)
```
Comment (merelcht) [inline kedro/framework/cli/catalog.py:236]: I'm aware that using `.get()` is inconsistent with how we get the catalog from the config loader, but there's a higher chance a user doesn't have any credentials. I'm also wondering if using `.get()` is problematic with how stuff works in python 3.12 (cc: @AhdraMeraliQB @ankatiyar)

**Review by DimedS — APPROVED** [review]
Thanks for the PR, @merelcht ! LGTM. I agree, using .get() here is a good choice.

File: kedro/framework/cli/catalog.py (hunk 1 above)
Comment (ankatiyar) [reply kedro/framework/cli/catalog.py:236]: I think using `.get()` should be fine with python 3.12 after #3587

**Review by ankatiyar — APPROVED** [review]
Thanks @merelcht!

**Review by AhdraMeraliQB — APPROVED** [review]

**Pull request merged**